Orange/data/table/base.py

```diff
@@ -0,0 +1,1292 @@
+            cls._next_instance_id += num
+            return out[0] if num == 1 and not force_list else out
+
+    def _transform_discrete_values_to_descriptors(self):
+        """Transform discrete variables given in index form into descriptors.
+
+        The conversion is performed in-place.
+        """
+        for var in chain(self.domain.variables, self.domain.metas):
+            if isinstance(var, DiscreteVariable):
+                # only transform the values if all of them appear to be integers and
+                # could act as a variable value index
+                # otherwise we're dealing with numeric discretes
+                is_values = self[var.name].apply(lambda v: isinstance(v, Number) and
+                                                           (isinstance(v, int) or v.is_integer()) and
+                                                           v < len(var.values)).all()
+                if is_values:
+                    self[var.name] = self[var.name].apply(lambda v: var.values[int(v)])
+
+    def _transform_discrete_into_categorical(self):
+        """Transform columns with discrete variables into pandas' categoricals.
+
+        The operation is performed in-place.
+        This must be done after replacing null values because those aren't values,
+        and also after transforming discretes to descriptors.
+        """
+        for var in chain(self.domain.variables, self.domain.metas):
+            if isinstance(var, DiscreteVariable):
+                self[var.name] = pd.Categorical(self[var.name], categories=var.values, ordered=var.ordered)
+
+    def _transform_timevariable_into_datetime(self):
+        """Transform columns with TimeVariables into pandas' datetime columns (in-place)."""
+        for var in chain(self.domain.variables, self.domain.metas):
+            if isinstance(var, TimeVariable):
+                self[var.name] = var.column_to_datetime(self[var.name])
+
+    def save(self, filename):
+        """Save a Table to a file.
+
+        Parameters
+        ----------
+        filename : str
+            The destination file path, absolute or relative.
+        """
+        ext = os.path.splitext(filename)[1]
+        from Orange.data.io import FileFormat
+        writer = FileFormat.writers.get(ext)
+        if not writer:
+            desc = FileFormat.names.get(ext)
+            if desc:
+                raise IOError("Writing of {}s is not supported".format(desc.lower()))
+            else:
+                raise IOError("Unknown file name extension.")
+        writer.write_file(filename, self)
+
+    def __getitem__(self, item):
+        # if selecting a column subset, we need to transfer weights so they don't just disappear
+        # only do this for multiple column selection, which returns a DataFrame by contract
+        if isinstance(item, (Sequence, pd.Index)) and not isinstance(item, str) \
+                and all(isinstance(i, str) for i in item) \
```

Comment:
Your pseudocode is concatenation right? With this being `__getitem__`, I think it's way more performant to pre-process the selected columns. Also, this would have to have another if that checks if it's a Series and stack that, and also that it's not a column (because we don't want to stack that).